lib/items/uploadSession.js

```diff
@@ -60,7 +60,7 @@ async function uploadSession(params, progress = () => {}) {
   // chunks we've accumulated in memory that we're going to upload
   let chunks = [];
 
-  const initGotExtended = got.extend({
+  const gotExtend = (await got).got.extend({
```

Comment:
following: is there anything else causing this?

Reply:
i guess there is some change in the got.stream method. I just didn't have a chance to look deeper into that

Reply:
@paradix

Reply:
Well guys for me the behavior is exactly opposite. When I revert I get the following result.

```
  17 passing (39s)
  1 failing

  1) download
       Should download Simple file using Stream:
     Error: Timeout of 15000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\__PRIVATE\_PROJEKTY\onedrive-api\test\e2e\items\download.test.js)
      at listOnTimeout (node:internal/timers:573:17)
      at process.processTimers (node:internal/timers:514:7)
```

Reply:
> Well guys for me the behavior is exactly opposite. When I revert I get the following result.
> 
> ```
>   17 passing (39s)
>   1 failing
> 
>   1) download
>        Should download Simple file using Stream:
>      Error: Timeout of 15000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\__PRIVATE\_PROJEKTY\onedrive-api\test\e2e\items\download.test.js)
>       at listOnTimeout (node:internal/timers:573:17)
>       at process.processTimers (node:internal/timers:514:7)
> ```

Somehow my Windows PC just fail the test, unlike your result. I tested it on my MacBook and it works perfectly, so your E2E result must be correct. The only difference between my 2 machines is my MacOS uses node `16.15.0`, while Windows is `18.14.0`, so I guess this is the reason then.

But still, about the fix I mentioned below:

>While this can fix most of it ![image](https://user-images.githubusercontent.com/29449869/228901660-c7d2f5bc-3cea-4685-b1ac-78e5fbbfb782.png)

I think you will need `done` to pass to the `errorHandler` (it's required and will be triggered as a function).

Reply:
I just commited a change in the download test. Have a look now. Should be fine now.